File: z-clients/javascript/src/apis/msgsSink.ts
// this file is @generated

import {
    type ListResponseSinkOut,
    ListResponseSinkOutSerializer,
} from '../models/listResponseSinkOut';
import {
    type SinkConfigureIn,
    SinkConfigureInSerializer,
} from '../models/sinkConfigureIn';
import {
    type SinkConfigureOut,
    SinkConfigureOutSerializer,
} from '../models/sinkConfigureOut';
import {
    type SinkDeleteIn,
    SinkDeleteInSerializer,
} from '../models/sinkDeleteIn';
import {
    type SinkDeleteOut,
    SinkDeleteOutSerializer,
} from '../models/sinkDeleteOut';
import {
    type SinkListIn,
    SinkListInSerializer,
} from '../models/sinkListIn';
import { HttpMethod, DiomRequest, type DiomRequestContext } from "../request";

export class MsgsSink {
    public constructor(private readonly requestCtx: DiomRequestContext) {}

    /** Create or update a sink for a topic. Overwrites any existing sink with the same id. */
    public configure(
        sinkConfigureIn: SinkConfigureIn,
    ): Promise<SinkConfigureOut> {
        const request = new DiomRequest(HttpMethod.POST, "/api/v1.msgs.sink.configure");

        request.setBody(
            SinkConfigureInSerializer._toJsonObject(sinkConfigureIn)
        );
        
        return request.send(
            this.requestCtx,
            SinkConfigureOutSerializer._fromJsonObject,
        );
    }/** Delete a sink. */
    public delete(
        sinkDeleteIn: SinkDeleteIn,
    ): Promise<SinkDeleteOut> {
        const request = new DiomRequest(HttpMethod.POST, "/api/v1.msgs.sink.delete");

        request.setBody(
            SinkDeleteInSerializer._toJsonObject(sinkDeleteIn)
        );
        
        return request.send(
            this.requestCtx,
            SinkDeleteOutSerializer._fromJsonObject,
        );
    }/** List sink configurations for a topic. */
    public list(
        topic: string,
        sinkListIn: SinkListIn,
    ): Promise<ListResponseSinkOut> {
        const request = new DiomRequest(HttpMethod.POST, "/api/v1.msgs.sink.list");

        request.setBody(
            SinkListInSerializer._toJsonObject({
                ...sinkListIn,
                topic: topic,
            })
        );
        
        return request.send(
            this.requestCtx,
            ListResponseSinkOutSerializer._fromJsonObject,
        );
    }
}

File: z-clients/java/src/main/java/com/svix/diom/apis/MsgsSink.java
// this file is @generated
package com.svix.diom.apis;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.svix.diom.DiomException;
import com.svix.diom.HttpClient;
import java.util.Collections;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.Set;
import com.svix.diom.models.ListResponseSinkOut;
import com.svix.diom.models.SinkConfigureIn;
import com.svix.diom.models.SinkConfigureOut;
import com.svix.diom.models.SinkDeleteIn;
import com.svix.diom.models.SinkDeleteOut;
import com.svix.diom.models.SinkListIn;
import com.svix.diom.models.SinkListIn_;

public class MsgsSink {
    private final HttpClient client;

    public MsgsSink(HttpClient client) {
        this.client = client;
    }

    /** Create or update a sink for a topic. Overwrites any existing sink with the same id. */
    public SinkConfigureOut configure(
        final SinkConfigureIn sinkConfigureIn
    ) throws DiomException {

        return this.client.executeRequest(
            "POST",
            "/api/v1.msgs.sink.configure",
            null,
            sinkConfigureIn,
            SinkConfigureOut.class
        );
    }

    /** Delete a sink. */
    public SinkDeleteOut delete(
        final SinkDeleteIn sinkDeleteIn
    ) throws DiomException {

        return this.client.executeRequest(
            "POST",
            "/api/v1.msgs.sink.delete",
            null,
            sinkDeleteIn,
            SinkDeleteOut.class
        );
    }

    /** List sink configurations for a topic. */
    public ListResponseSinkOut list(
        String topic,
        final SinkListIn sinkListIn
    ) throws DiomException {
        SinkListIn_ body = new SinkListIn_(
            sinkListIn.getNamespace(),
            topic,
            sinkListIn.getLimit(),
            sinkListIn.getIterator()
        );

        return this.client.executeRequest(
            "POST",
            "/api/v1.msgs.sink.list",
            null,
            body,
            ListResponseSinkOut.class
        );
    }

    /** List sink configurations for a topic. */
    public ListResponseSinkOut list(
        String topic
    ) throws DiomException {
        return this.list(
            topic,
            new SinkListIn()
        );
    }
}